src/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useEffect, useState } from 'react';
+
+type CopiedValue = string | null;
+type CopyFn = (text: string) => Promise<boolean>;
+
+export const useCopyToClipboard = (ms = 500): [CopiedValue, CopyFn, boolean] => {
+    const [copiedText, setCopiedText] = useState<CopiedValue>(null);
+    const [isSuccess, setIsSuccess] = useState(false);
+
+    useEffect(() => {
+        let timer: NodeJS.Timer;
+        if (isSuccess) {
+            timer = setTimeout(() => {
+                setIsSuccess(false);
+            }, ms);
+        }
+
+        return () => {
+            window.clearTimeout(timer);
+        };
+    }, [isSuccess, ms]);
+
+    const copy: CopyFn = useCallback(async (text) => {
+        if (!navigator?.clipboard) {
+            const clipboardNotSupportedError = new Error('Clipboard not supported');
+            console.warn(clipboardNotSupportedError);
+            return Promise.reject(clipboardNotSupportedError);
```

Comment:
I think it would be better to recognize it earlier and hide copy button instead of provoke user to get error